src/main/resources/META-INF/resources/primefaces/datatable/datatable.js

```diff
@@ -1125,20 +1125,23 @@ PrimeFaces.widget.DataTable = PrimeFaces.widget.DeferredWidget.extend({
         }
 
         //keyboard support
-        this.tbody.off('focus.dataTable blur.dataTable change.dataTable', checkboxSelector)
-                    .on('focus.dataTable', checkboxSelector, null, function() {
+        this.tbody.off('focus.dataTableCheck blur.dataTableCheck change.dataTableCheck click.dataTableCheck', checkboxSelector)
+                    .on('focus.dataTableCheck', checkboxSelector, null, function() {
                         var input = $(this);
 
-                        if(input.attr('aria-checked') === "true" || input.prop('checked')) {
-                            input.removeClass('ui-state-active');
-                        }
```

Comment:
This part above makes absolutely no sense to remove the active state just because the checkbox has focus.